appium/webdriver/extensions/gsm.py

```diff
@@ -46,7 +46,7 @@ def make_gsm_call(self, phone_number, action):
         :Usage:
             self.driver.make_gsm_call('5551234567', GsmCallActions.CALL)
         """
-        constants = get_dict_const_attr_and_val(GsmCallActions)
+        constants = extract_const_attributes(GsmCallActions)
         if action not in constants.values():
```

Comment:
You need an ordered dict there or array of pairs, since the order of keys and values is not guaranteed to be stable in the default dict implementation

Reply:
Hum, could you tell me why ordered dict is necessary here?

> the order of keys and values is not guaranteed to be stable in the default dict implementation

I see this behavior.

Reply:
https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6/39980744

Reply:
means the actual behaviour depends on the Python version, so I wouldn't rely on it

Reply:
Thanks, I didn't know details from impl point of view.
I mean, what does it happen if dict order is not guaranteed?

My understanding is as below.

1. No impact to the results
   *  both ```'call' not in ['call', 'accept']``` and ```'call' not in ['accept', 'call']``` have the same result
2. Warning log can be changed along to python version
   * ```Consider using one of [CALL, ACCEPT]``` and ```Consider using one of [ACCEPT, CALL]``` can show up in user env.
   * If this order should be fixed, I'll fix dict order.

If my understanding is short, let me know.

Reply:
for this particular case it does not matter, but for other things it might be good to keep the original order when showing the list of available options. You could simply use OrderedDict from `collections` to ensure such functionality without changing the client methods